src/Workspaces/Core/Portable/Workspace/WorkspaceEventOptions.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis;
+
+internal record struct WorkspaceEventOptions(bool RequiresMainThread)
+{
+    public static readonly WorkspaceEventOptions Default = new(RequiresMainThread: false);
+    public static readonly WorkspaceEventOptions MainThreadDependent = new(RequiresMainThread: true);
```

Comment:
```suggestion
    public static readonly WorkspaceEventOptions RequiresMainThread = new(RequiresMainThread: true);
```

WorkspaceEventOptions.RequiresMainThread reads very well to me (vs MainThreadDependent).